pkg/util/allocationfilterutil/queryfilters.go

```diff
@@ -289,7 +289,7 @@ func filterV1LabelMappedFromList(rawFilterValues []string, labelName string) kub
 		filterValue, wildcard := parseWildcardEnd(filterValue)
 
 		subFilter := kubecost.AllocationFilterCondition{
-			Field: kubecost.FilterLabel,
+			Field: kubecost.FilterAlias,
```

Comment:
I'd like to suggest a different approach here. If you look at the comment on line 26 of allocationfilter.go, you'll see this:

```
// Filtering based on label aliases (team, department, etc.) should be a
// responsibility of the query handler. By the time it reaches this
// structured representation, we shouldn't have to be aware of what is
// aliased to what.
```

I made this very intentionally so that filters wouldn't need to have any concept of aliasing. Instead, I think a slightly different approach could work here:

The `subFilter` could be `(or (= label x) (= annotation y))`. 

Does that make sense?

Reply:
Sorry, just realized this is the cherry pick. Moved to main PR: https://github.com/opencost/opencost/pull/1457#discussion_r1006354483

Reply:
At least, it's the only approach we could come up with on the spot. It's really a case of a regrettable product decision. Consider the case of:

* filtering on `team == "kubecost"`
* allocation with `label[team] = "monitoring"` and `annotation[team] = "kubecost"`

Nick's logic would return `false`, but your proposed change would return `true`.